pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -146,6 +146,28 @@ func (r *SystemReconciler) Reconcile() (reconcile.Result, error) {
 	// Used to synchronize rollout of system Deployments
 	systemComponentNotReady := false
 
+	// If the image has changed, delete the PreHook/PostHook Jobs so they can be recreated with the new image
+	imageChanged, err := helper.HasJobImageChanged(component.SystemAppPreHookJobName, r.apiManager.GetNamespace(), ampImages.Options.SystemImage, r.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if imageChanged {
```

Comment:
@carlkyrillos , should we check if job is running before delete it?. If job is completed or even failed - suppose we can delete it. But if running - maybe not (?). But if we let's say that Job status is not interesting (like in current code, if I'm understand) so we again return to question - maybe no need delete it in operator and allow automatic deletion ..(?)

Reply:
There's no harm in adding a check to make sure the Job isn't running before deleting it. I'll fix this and push the changes. Thanks @valerymo